app.rb

```diff
@@ -1,10 +1,141 @@
 require 'sinatra'
 require 'active_record'
 require 'sqlite3'
+require 'pry'
+require 'ostruct'      #Enables use of OpenStruct
+require_relative 'models/note'
 
 ActiveRecord::Base.establish_connection(
   adapter:  'sqlite3',
   database: 'db/development.sqlite3'
 )
 
 
+get '/api/notes.json' do
+  notes = Note.all
+  notes_with_tags = notes.collect do |x|
+    x.example_note
+  end
+  notes_with_tags.to_json
+end
+
+get '/api/notes_long_way.json' do
+  notes = Note.all
+  notes_with_tags = notes.collect do |x|
+    { title:  x.title,
+      body:   x.body,
+      tags:   x.tags.collect do |z| { name: z.name } end
+    }
+  end
+  notes_with_tags.to_json
+end
+
+# #Returns the given tag with all it's related notes, first pass
+# get '/api/notes/tag/:name' do
+#   tag = Tag.find_by(name: params[:name])
+#   tag_with_notes =
+#       { name:  tag.name,
+#         notes: tag.notes.collect do |a|
+#                { title: a.title,
+#                  body:  a.body,
+#                  tags:  a.tags.collect do |z|
+#                    { name: z.name }
+#                  end
+#                      }
+#         end
+#       }
+#   tag_with_notes.to_json
+# end
+
+#Returns the given tag with all it's related notes
+get '/api/notes/tag/:name' do
+  tag = Tag.find_by(name: params[:name])
+  tag_with_notes =
+      { name:  tag.name,
+        notes: tag.notes.collect do |note|
+                note.hash_format_title_body_tags
+        end
+      }
+  tag_with_notes.to_json
+end
+
+
+
+
+post '/api/notes' do
+
+  new_note = Note.create(title: params[:title],
+                        body: params[:body])
+  if new_note.save
+    array = params[:tags].split(",")
+    array.each do |x|
+      new_tag = Tag.create!(name: x)
```

Comment:
This wouldn't work if Jacques was reusing a tag name.

It's also not stripping out the extra spaces.